### PR TITLE
360 create dockerfile that provisions the node server

### LINKED ISSUE #360: Create DockerFile that provisions Server Node

### PR DESCRIPTION
**Pull request by Elkrival**
This pr provisions the dockerfile for the node server, it also removes loading configurations from config file written by scripts and instead uses env variables and constants to load the settings.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,6 +15,7 @@ node_modules
 config.js
 .env
 .env.development
+docker-compose.yml
```
Comment (ngmaloney) [inline .gitignore]: What is the rationale for not committing this and having a compose template? Ideally, a developer can clone the repo and everything (mostly) "just works" by running `docker-compose up` but I might be missing something.

Comment (Elkrival) [reply .gitignore]: The path where I store the test data and backups is not the usual path for mongo. In the docker-compose file the developer has to add the path to their data that will seed the mongo service. By copying the template they have everything and just update the path.

Comment (Elkrival) [reply .gitignore]: @ngmaloney [dce03ce](https://github.com/AMP-SCZ/dpdash/pull/367/commits/dce03cefbd6186c4d70ddf944c5497013bec23ab)

File: server/app.js [hunk 2]
```diff
@@ -27,13 +26,16 @@ import dashboardsRouter from './routes/dashboards'
...
-import config from './configs/config'
 import basePathConfig from './configs/basePathConfig'
+import { PASSPORT_FIELDS_ATTRIBUTES } from './constants'
 
 const localStrategy = Strategy
 
 const basePath = basePathConfig || ''
-
+const cookieAttributes = {
+  secure: false,
```
Comment (JackVCurtis) [inline server/app.js]: Do we want this all the time? Is this overridden somewhere?

File: server/app.js [hunk 3]
```diff
@@ -110,7 +112,10 @@ app.use(methodOverride())
 let mongodb
 const mongoURI = process.env.MONGODB_URI
 const mongodbPromise = co(function* () {
-  return yield MongoClient.connect(mongoURI, config.database.mongo.server)
+  return yield MongoClient.connect(mongoURI, {
+    ssl: false,
```
Comment (JackVCurtis) [inline server/app.js]: Same here, shouldn't this only be in development?

File: docker-compose.yml [hunk 4]
```diff
@@ -21,4 +19,4 @@ services:
       MONGO_INITDB_ROOT_USERNAME: $MONGODB_ADMIN_USER
       MONGO_INITDB_ROOT_PASSWORD: $MONGODB_ADMIN_PASSWORD
     volumes:
-      - <path-to-your-/data/db>:/data/db
+      - $MONGO_DATA_PATH:/data/db
```
Comment (JackVCurtis) [inline docker-compose.yml]: Does this have to be a path? I'm using a Docker managed volume currently

**Comment by JackVCurtis** [comment]
Do we have to clean up the generator for `config.js` so it doesn't create irrelevant config variables that now come from the environment? If we're moving away from the generator that's probably not necessary, but if we still have to run `./init.sh` we should probably clean up unused variables.

File: server/app.js (hunk 3 above)
Comment (Elkrival) [reply server/app.js]: 👍 
[c2c8c19](https://github.com/AMP-SCZ/dpdash/pull/367/commits/c2c8c19e9583c3411f0036557c60bc1b65bc357a)

File: server/app.js (hunk 2 above)
Comment (Elkrival) [reply server/app.js]: 👍 
[c2c8c19](https://github.com/AMP-SCZ/dpdash/pull/367/commits/c2c8c19e9583c3411f0036557c60bc1b65bc357a)

File: docker-compose.yml (hunk 4 above)
Comment (Elkrival) [reply docker-compose.yml:22]: @JackVCurtis I'm thinking on how to make this easier for someone to set up. I imagine we would need a mongodump of the data and import it. Was your volume set up before running mongo in docker..?

**Comment by Elkrival** [comment]
> Do we have to clean up the generator for `config.js` so it doesn't create irrelevant config variables that now come from the environment? If we're moving away from the generator that's probably not necessary, but if we still have to run `./init.sh` we should probably clean up unused variables.

@JackVCurtis  so we will be removing all of those scripts https://github.com/AMP-SCZ/dpdash/issues/354

File: docker-compose.yml (hunk 4 above)
Comment (JackVCurtis) [reply docker-compose.yml]: Yeah I ran the mongo container independent of docker-compose to import the data dump.

**Review by JackVCurtis — APPROVED** [review]
👍 LGTM

File: docker-compose.yml (hunk 4 above)
Comment (Elkrival) [reply docker-compose.yml]: 👍 Thanks for that. Made the change f1f0b7481f8ed6ba80418e19b6c66830bc2009a4

**Pull request merged**